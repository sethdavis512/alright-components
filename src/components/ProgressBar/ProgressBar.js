import React, { Component } from 'react';
import PropTypes from 'prop-types';

class ProgressBar extends Component {
    getColor = () => {
        if (this.props.percent === 100) return 'green';
        return this.props.percent > 50
            ? 'gold'
            : this.props.percent > 25
            ? 'red'
            : 'firebrick';
    };

    getWidthAsPercentOfTotalWidth = () => {
        return parseInt(this.props.width * (this.props.percent / 100), 10);
    };

    render() {
        const { percent, width, height } = this.props;
        return (
            <div
                style={{
                    border: '1px solid lightgray',
                    width,
                    borderRadius: '4px'
                }}
            >
                <div
                    style={{
                        width: this.getWidthAsPercentOfTotalWidth(),
                        height,
                        backgroundColor: this.getColor(percent)
                    }}
                />
            </div>
        );
    }
}

ProgressBar.defaultProps = {
    height: 20,
    width: 200
};

ProgressBar.propTypes = {
    percent: PropTypes.number.isRequired,
    width: PropTypes.number.isRequired,
    height: PropTypes.number
};

export default ProgressBar;
