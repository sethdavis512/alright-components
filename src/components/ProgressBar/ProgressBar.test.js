import React from 'react';
import { shallow } from 'enzyme';
import ProgressBar from './ProgressBar';

describe('ProgressBar', () => {
    const defaultProps = {
        percent: 50
    };

    function renderComponent(props) {
        return shallow(<ProgressBar {...defaultProps} {...props} />);
    }

    it('should getWidthAsPercentOfTotalWidth of 250 with total width of 500 and percent of 50', () => {
        const wrapper = renderComponent({ width: 500 });
        const width = wrapper.instance().getWidthAsPercentOfTotalWidth();
        expect(width).toEqual(250);
    });

    it('should match default snapshot', () => {
        const wrapper = renderComponent();
        expect(wrapper).toMatchSnapshot();
    });
});
