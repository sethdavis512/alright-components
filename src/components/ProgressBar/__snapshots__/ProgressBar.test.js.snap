// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar should match default snapshot 1`] = `
<div
  style={
    Object {
      "border": "1px solid lightgray",
      "borderRadius": "4px",
      "width": 200,
    }
  }
>
  <div
    style={
      Object {
        "backgroundColor": "red",
        "height": 20,
        "width": 100,
      }
    }
  />
</div>
`;
